gluten-core/src/main/scala/io/glutenproject/expression/QuaternaryExpressionTransformer.scala

```diff
@@ -24,6 +24,42 @@ import io.glutenproject.substrait.expression.{ExpressionBuilder, ExpressionNode}
 
 import org.apache.spark.internal.Logging
 import org.apache.spark.sql.catalyst.expressions.Expression
+import org.apache.spark.sql.catalyst.expressions.RegExpReplace
+import io.glutenproject.backendsapi.BackendsApiManager
+import io.glutenproject.GlutenConfig
+import org.apache.spark.sql.catalyst.expressions.Literal
+import org.apache.spark.sql.types.IntegerType
+import io.glutenproject.substrait.expression.IntLiteralNode
+
+class RegExpReplaceTransformer(substraitExprName: String, subject: ExpressionTransformer,
+  regexp: ExpressionTransformer, rep: ExpressionTransformer, pos: ExpressionTransformer,
+  original: RegExpReplace)
+  extends ExpressionTransformer with Logging {
+
+  override def doTransform(args: java.lang.Object): ExpressionNode = {
+    if (BackendsApiManager.getBackendName.equalsIgnoreCase(GlutenConfig.GLUTEN_VELOX_BACKEND)) {
+      QuaternaryExpressionTransformer(ExpressionMappings.REGEXP_REPLACE,
+        subject, regexp, rep, pos, original).doTransform(args)
+    }
+
+
```

Comment:
One empty line is enough, is there any other difference between QuaternaryExpressionTransformer and QuaternaryExpressionTransformer except posNode? Could you merge two into one?